src/riscv/linux/riscv-hw.c

```diff
@@ -16,10 +21,6 @@ void cpuinfo_riscv_linux_decode_vendor_uarch_from_hwprobe(
 	};
 	const size_t pairs_count = sizeof(pairs) / sizeof(struct riscv_hwprobe);
 
-	/* In case of failure, report unknown. */
-	*vendor = cpuinfo_vendor_unknown;
-	*uarch = cpuinfo_uarch_unknown;
-
```

Comment:
If we're going to remove this, can we change this function to report a bool success or failure? The caller should know whether or not to trust the information in the structure.